proxy/healthy_endpoints_test.go

```diff
@@ -165,41 +179,51 @@ func newRoundTripperUnhealthyHost(o *RoundTripperUnhealthyHostOptions) func(r ht
 }
 
 func TestPHCForMultipleHealthyAndOneUnhealthyEndpoints(t *testing.T) {
-	services := []*httptest.Server{}
-	for i := 0; i < 3; i++ {
-		service := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			w.WriteHeader(http.StatusOK)
-		}))
-		services = append(services, service)
-		defer service.Close()
-	}
-	endpointRegistry := defaultEndpointRegistry()
-
-	doc := fmt.Sprintf(`* -> <random, "%s", "%s", "%s">`, services[0].URL, services[1].URL, services[2].URL)
-	tp, err := newTestProxyWithParams(doc, Params{
-		EnablePassiveHealthCheck:   true,
-		EndpointRegistry:           endpointRegistry,
-		CustomHttpRoundTripperWrap: newRoundTripperUnhealthyHost(&RoundTripperUnhealthyHostOptions{Host: services[0].URL[7:], Probability: rtFailureProbability}),
-	})
-	if err != nil {
-		t.Fatal(err)
-	}
-	defer tp.close()
-
-	ps := httptest.NewServer(tp.proxy)
-	defer ps.Close()
-
-	failedReqs := 0
-	for i := 0; i < nRequests; i++ {
-		rsp, err := ps.Client().Get(ps.URL)
-		if err != nil {
-			t.Fatal(err)
-		}
-
-		if rsp.StatusCode != http.StatusOK {
-			failedReqs++
-		}
-		rsp.Body.Close()
+	for _, algorithm := range []string{"random", "consistentHash", "roundRobin", "powerOfRandomNChoices"} {
+		t.Run(algorithm, func(t *testing.T) {
+			services := []*httptest.Server{}
+			for i := 0; i < 3; i++ {
```

Comment:
I don't see the unhealthy endpoint

Reply:
This unhealthy endpoint was emulated by custom round tripper before.
I see this custom round tripper is not understandable for other people, so I decided to add backendTimeout filter to the skipper route and add corresponding `time.Sleep` to the endpoint which is emulated to be "unhealthy".

Reply:
The tests are still failing before fix and passing after fix.